### PR TITLE
Custom embed schema implementation

### PR DESCRIPTION
**Pull request by gengel**
Implementation of #205

### CODE REVIEW

File: src/main/resources/schema/ans/0.10.0/story_elements/custom_embed.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        },
+
+        "config": {
+          "title": "Embed Configuration",
+          "description": "Arbitrary configuration data generated by a plugin. Users are responsible for maintaining schema.",
+          "type": "object",
+
+          "additionalProperties": false,
+          "patternProperties": {
+            "^(?!(referent|type)$)[a-zA-Z0-9_]*$": {}
```
Comment (yury-sannikov) [inline src/main/resources/schema/ans/0.10.0/story_elements/custom_embed.json]: what about `version` field or it's okay to have it?

Comment (gengel) [reply src/main/resources/schema/ans/0.10.0/story_elements/custom_embed.json]: @yury-sannikov Good catch; just updated it

File: src/main/resources/schema/ans/0.10.0/story_elements/custom_embed.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          "type": "string",
+          "maxLength": 128,
+          "minLength": 1
+        },
+
+        "url": {
+          "title": "Embed Provider URL",
+          "description": "Provider URL for this embed item. When concatenated with Embed ID, should produce a URL that returns json metadata about the embedded content.",
+          "type": "string",
+          "maxLength": 128,
```
Comment (chenghsinw) [inline src/main/resources/schema/ans/0.10.0/story_elements/custom_embed.json]: I feel 128 length might not be long enough to cover most of the cases:
https://111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111111
Should we change it to 255?

Comment (gengel) [reply src/main/resources/schema/ans/0.10.0/story_elements/custom_embed.json]: Good catch! Looks like I copy+pasted id without thinking. I've bumped to 512

**Pull request merged**